### ISSUE TITLE
notify-osd cannot send message

### ISSUE THREAD

**Issue by memberbetty** [NixOS/nixpkgs#55817]
## Issue description

> GLib-GIO-Message: 11:25:13.647: Using the 'memory' GSettings backend.  Your settings will not be saved or shared with other applications.
> 
> ** (notify-osd:16786): WARNING **: 11:25:13.648: Another instance has already registered org.freedesktop.Notifications
> 
> ** (notify-osd:16786): WARNING **: 11:25:13.648: Could not register instance
> 

### Steps to reproduce
Desktop environment plasma5.

```
notify-osd a # not a typo
```

## Technical details

- system: `"x86_64-linux"`
 - host os: `Linux 4.14.98, NixOS, 18.09.git.168cbb3 (Jellyfish)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.1.3`
 - channels(root): `"nixos-18.09.1571.7795a7ad5f0"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

**Comment by hedning** [comment]
Try `notify-send` from `libnotify`, I'm fairly certain `notify-osd` is a daemon which handles showing notifications (if I understood the question at least).

**Comment by jtojnar** [comment]
Yup. And I would guess that Plasma already has its own notification server that occupies the `org.freedesktop.Notifications` interface. You will need to find a way to disable Plasma’s notification server if you want to use `notify-osd` in its stead.

**Comment by jtojnar** [comment]
The first warning is from no GSettings module being available, see https://github.com/NixOS/nixpkgs/pull/43150 for more information.

**Comment by memberbetty** [comment]
Is it possible that we get output like "Copy paste the following to  configuration.nix on line X, column Y" in order to have a more helpful error message? 

(This would apply to many packages and probably requires a lot of patching, so I know it's not "free").

**Comment by jtojnar** [comment]
Patching [this](https://gitlab.gnome.org/GNOME/glib/blob/aa6aa8b5e4547d688748d2bbb9634f6575c1789c/gio/gsettingsbackend.c#L980-981) should be enough and the issue can be fixed by adding `programs.dconf.enable = true;` to `configuration.nix`.

Fixing it will unlock [this](https://gitlab.gnome.org/GNOME/glib/blob/aa6aa8b5e4547d688748d2bbb9634f6575c1789c/gio/gsettings.c#L590) or [this](https://gitlab.gnome.org/GNOME/glib/blob/aa6aa8b5e4547d688748d2bbb9634f6575c1789c/gio/gsettings.c#L595) which requires much more involved fix – either adding `wrapGAppsHook` to dependencies of the application, or patching the schema path into the library causing the issue (necessary for plug-ins). We should probably first finish https://github.com/NixOS/nixpkgs/pull/43150 and link there from the messages once it is done.
